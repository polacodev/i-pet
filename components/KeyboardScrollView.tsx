import React, { ReactNode } from 'react';
import { StyleSheet } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

type KeyboardScrollViewProps = {
  children: ReactNode;
};

const KeyboardScrollView = ({ children }: KeyboardScrollViewProps) => {
  return (
    <KeyboardAwareScrollView
      style={styles.container}
      resetScrollToCoords={{ x: 0, y: 0 }}
      contentContainerStyle={styles.scrollView}
      scrollEnabled>
      {children}
    </KeyboardAwareScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
  },
  scrollView: {
    flexGrow: 1,
  },
});

export default KeyboardScrollView;
