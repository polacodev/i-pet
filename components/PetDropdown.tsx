import React from 'react';
import { StyleSheet, type TextInputProps } from 'react-native';
import { Dropdown } from 'react-native-element-dropdown';

import { PetView } from './PetView';

import { BORDER_COLOR_FORM } from '@/constants/Colors';
import { useThemeColor } from '@/hooks/useThemeColor';
import { DropdownProps } from '@/types/dropdown';

type PetDropdownProps = TextInputProps & {
  lightColor?: string;
  darkColor?: string;
  onChange: (text: string) => void;
  value: string;
  data: DropdownProps[];
  placeholder: string;
};

export const PetDropdown = ({
  lightColor,
  darkColor,
  onChange,
  value,
  data,
  placeholder,
}: PetDropdownProps) => {
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');
  const placeholderColor = useThemeColor({ light: lightColor, dark: lightColor }, 'placeholder');
  const borderColorForm = BORDER_COLOR_FORM;

  return (
    <PetView style={styles.container}>
      <Dropdown
        testID="pet-dropdown"
        style={[
          styles.dropdown,
          { borderColor: borderColorForm, borderWidth: 1, borderRadius: 10 },
        ]}
        placeholderStyle={[{ color: placeholderColor }, styles.placeholderStyle]}
        selectedTextStyle={[{ color }, styles.selectedTextStyle]}
        data={data}
        maxHeight={300}
        labelField="label"
        valueField="value"
        placeholder={placeholder}
        value={value}
        onChange={(item) => {
          onChange(item.value);
        }}
      />
    </PetView>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '85%',
  },
  dropdown: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 0.5,
    borderRadius: 8,
    paddingHorizontal: 8,
  },
  placeholderStyle: {
    paddingLeft: 17,
    fontSize: 14,
    fontWeight: 'normal',
  },
  selectedTextStyle: {
    paddingLeft: 17,
    fontSize: 14,
  },
});
