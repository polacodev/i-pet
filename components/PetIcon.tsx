import Ionicons from '@expo/vector-icons/Ionicons';
import React from 'react';
import { type ImageProps } from 'react-native';

type PetIconProps = ImageProps & {
  name?:
    | 'stop'
    | 'trash'
    | 'add-circle'
    | 'heart'
    | 'star'
    | 'camera'
    | 'logo-whatsapp'
    | 'close-circle'
    | 'save'
    | 'list'
    | 'paw'
    | 'person'
    | 'information-circle'
    | 'albums'
    | 'home'
    | 'qr-code'
    | 'log-in'
    | 'log-out';
  color?: string;
  size?: number;
  onPress?: () => void;
};

export const PetIcon: React.FC<PetIconProps> = ({
  style,
  name = 'stop',
  color = '#ffffff',
  size = 30,
  onPress,
  ...rest
}) => {
  return (
    <Ionicons
      testID="pet-icon"
      name={name}
      size={size}
      color={color}
      style={[style]}
      onPress={onPress}
      {...rest}
    />
  );
};
