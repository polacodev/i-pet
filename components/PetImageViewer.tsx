import React from 'react';
import { StyleSheet, Image, ImageSourcePropType } from 'react-native';

type PetImageViewerProps = {
  placeholderImageSource: ImageSourcePropType;
  selectedImage: string | null;
};

export const PetImageViewer: React.FC<PetImageViewerProps> = ({
  placeholderImageSource,
  selectedImage,
}) => {
  const imageSource = selectedImage ? { uri: selectedImage } : placeholderImageSource;
  return <Image testID="pet-image" source={imageSource} style={styles.image} />;
};

const styles = StyleSheet.create({
  image: {
    width: 80,
    height: 80,
    borderRadius: 50,
  },
});
