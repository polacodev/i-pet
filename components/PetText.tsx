import { Text, type TextProps, StyleSheet } from 'react-native';

import { useThemeColor } from '@/hooks/useThemeColor';

export type PetTextProps = TextProps & {
  lightColor?: string;
  darkColor?: string;
  type?: 'default' | 'smallText';
};

export const PetText = ({
  style,
  lightColor,
  darkColor,
  type = 'default',
  ...rest
}: PetTextProps) => {
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  let textType = undefined;
  switch (type) {
    case 'default':
      textType = styles.default;
      break;
    case 'smallText':
      textType = styles.smallText;
      break;
  }

  return <Text testID="pet-text" style={[{ color }, textType, style]} {...rest} />;
};

const styles = StyleSheet.create({
  default: {
    fontSize: 16,
    lineHeight: 24,
  },
  smallText: {
    fontSize: 14,
    lineHeight: 20,
  },
});
