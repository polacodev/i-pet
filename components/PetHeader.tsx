import { router } from 'expo-router';

import { PetIcon } from './PetIcon';
import { PetTitle } from './PetTitle';
import { PetView } from './PetView';
import { useUser } from './context/UserContext';

export const PetHeader = () => {
  const { session, user } = useUser();

  const goToLogin = () => {
    router.navigate('/login');
  };

  const goToProfile = () => {
    router.navigate('/profile');
  };

  return (
    <PetView>
      {session?.user ? (
        <PetIcon onPress={goToProfile} name="person" size={24} color="#0891b2" />
      ) : (
        <PetTitle type="link" onPress={goToLogin}>
          Log In
        </PetTitle>
      )}
    </PetView>
  );
};
