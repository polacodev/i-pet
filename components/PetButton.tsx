import React from 'react';
import { TouchableOpacity } from 'react-native';

import { PetIcon } from './PetIcon';
import { PetText } from './PetText';

type PetButtonProps = {
  onPress: () => void;
  iconName: 'heart' | 'star' | 'camera' | 'whatsapp' | 'save';
  buttonName: string;
  color?: '#0e7490' | '#075E54';
};

export const PetButton: React.FC<PetButtonProps> = ({
  onPress,
  iconName,
  buttonName,
  color = '#0e7490',
}) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: color,
        borderRadius: 8,
        padding: 8,
        gap: 8,
        width: '80%',
        margin: 6,
      }}>
      <PetIcon name={iconName} size={30} color="#ffffff" />
      <PetText type="default" style={{ color: '#ffffff' }}>
        {buttonName}
      </PetText>
    </TouchableOpacity>
  );
};
