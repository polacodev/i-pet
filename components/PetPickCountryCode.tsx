import { useEffect, useState } from 'react';
import { TouchableOpacity } from 'react-native';
import { CountryPicker } from 'react-native-country-codes-picker';

import { PetText } from './PetText';
import { PetView } from './PetView';

import { BORDER_COLOR_FORM } from '@/constants/Colors';

type PickProps = {
  onChange: (text: string) => void;
  value: string;
};

export const PetPickCountryCode = ({ onChange, value }: PickProps) => {
  const borderColorForm = BORDER_COLOR_FORM;
  const defaultCountryCode: string = '+1';
  const defaultCountryFlag: string = 'ðŸ‡ºðŸ‡¸';

  const [show, setShow] = useState(false);
  const [countryCode, setCountryCode] = useState(defaultCountryCode);
  const [countryFlag, setCountryFlag] = useState(defaultCountryFlag);

  useEffect(() => {
    onChange(defaultCountryCode);
    value = defaultCountryCode;
  }, []);

  return (
    <PetView style={{ width: '30%', height: 45, paddingRight: 2 }}>
      <TouchableOpacity
        testID="pet-pick-country"
        onPress={() => setShow(true)}
        style={{
          flex: 1,
          alignItems: 'center',
          justifyContent: 'center',
          borderWidth: 1,
          borderTopLeftRadius: 10,
          borderBottomLeftRadius: 10,
          borderColor: borderColorForm,
        }}>
        <PetText
          style={{
            fontSize: 14,
          }}>
          {countryFlag} {countryCode}
        </PetText>
      </TouchableOpacity>
      <CountryPicker
        style={{ modal: { height: '90%' } }}
        show={show}
        lang="en"
        pickerButtonOnPress={(item) => {
          value = item.dial_code;
          onChange(item.dial_code);
          setCountryCode(item.dial_code);
          setCountryFlag(item.flag);
          setShow(false);
        }}
      />
    </PetView>
  );
};
