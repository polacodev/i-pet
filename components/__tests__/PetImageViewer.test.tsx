import { render, screen } from '@testing-library/react-native';
import '@testing-library/react-native/extend-expect';

import { PetImageViewer } from '@/components/PetImageViewer';

describe('<PetImageViewer />', () => {
  const placeholderImageDark = require('../../assets/images/pet-image-dark.png');
  const placeholderImageLight = require('../../assets/images/pet-image-light.png');

  test('Image renders correctly on PetImageViewer', () => {
    const selectedImage = 'test.png';
    render(
      <PetImageViewer
        selectedImage={selectedImage}
        placeholderImageSource={placeholderImageLight}
      />,
    );

    const image = screen.getByTestId('pet-image');
    expect(image.props.source).toEqual({ uri: selectedImage });
  });

  test('Image styles applied correctly on PetImageViewer', () => {
    const selectedImage = '';
    const styles = { image: { width: 80, height: 80, borderRadius: 50 } };

    render(
      <PetImageViewer
        selectedImage={selectedImage}
        placeholderImageSource={placeholderImageDark}
      />,
    );

    const image = screen.getByTestId('pet-image');
    expect(image.props.style).toMatchObject(styles.image);
  });

  test('PetImageViewer renders correctly Snapshot', () => {
    const selectedImage = 'test.png';
    const tree = render(
      <PetImageViewer
        selectedImage={selectedImage}
        placeholderImageSource={placeholderImageDark}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
