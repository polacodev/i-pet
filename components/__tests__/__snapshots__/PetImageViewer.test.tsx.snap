// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PetImageViewer /> PetImageViewer renders correctly Snapshot 1`] = `
<Image
  source={
    {
      "uri": "test.png",
    }
  }
  style={
    {
      "borderRadius": 50,
      "height": 80,
      "width": 80,
    }
  }
  testID="pet-image"
/>
`;
